### PR TITLE
Account nickname is always defined

### PR DESCRIPTION
**Pull request by tommasini**
**Description**
The issue is that when the nickname of the account would be empty, it would be non editable.

**Proposed Solution**
The proposed solution it is, when the name it's empty we will assume that the name of the account will be the last name of it.
Example: 
If it was "Account 2" after confirm the empty string on the account name, it will come back to be "Account 2"

Also its proposed, for the users that already have no nickname on their accounts we set the default name "Account" when the user opens the app. 

**Code Impact**
Low

**Test Cases**
Case1:
* Notice the account nickname
* Long press the account nickname
* Edit the name to an empty string
* Will came back to the last account nickname

**Screenshots/Recordings**

https://user-images.githubusercontent.com/46944231/179572818-7b5589f9-b820-48a9-bc10-692f2213229b.mp4

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #4308 

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by jakehaugen** [comment]
Looks great to me!

File: app/components/UI/AccountOverview/index.js [hunk 1]
```diff
@@ -245,6 +245,13 @@ class AccountOverview extends PureComponent {
     InteractionManager.runAfterInteractions(() => {
       this.doENSLookup();
     });
+
+    //This is to fix the accounts which doesn't had name on their accounts
+    //If this is on Production for some time can be removed.
+    const { PreferencesController } = Engine.context;
+    if (!accountLabel) {
+      PreferencesController.setAccountLabel(selectedAddress, 'Account');
+    }
```
Comment (gantunesr) [inline app/components/UI/AccountOverview/index.js:254]: I think the code is self-explanatory, you can remove the comment

**Review by gantunesr — COMMENTED** [review]
LGTM. Good work, just a small comment

File: app/components/UI/AccountOverview/index.js (hunk 1 above)
Comment (tommasini) [reply app/components/UI/AccountOverview/index.js:254]: Okok! Doing it

**Review by gantunesr — APPROVED** [review]

**Review by cortisiko — CHANGES_REQUESTED** [review]
@tommasini I found a fun edge case:

If I were to:
- long press to edit my account nickname
- remove the current name
- tap on the space bar to set ' ' as my account nickname
- upon doing so, I still experience the bug where I am unable to edit my account nickname after setting it to ' '

see [recording](http://recordit.co/2Ngj9SyMVH)

**Comment by tommasini** [comment]
@cortisiko Really well catched, I was not considering the blank spaces!! Thank you Curtis! Awesome!!!

@gantunesr Could you give me another review please?

**Review by gantunesr — APPROVED** [review]

**Review by cortisiko — APPROVED** [review]
🌮 🌮 looks good @tommasini

**Pull request merged**